### ISSUE TITLE
Not working 

### ISSUE THREAD

**Issue by Cinichi** [f-o/MEGA-Account-Generator#13]
![Screenshot_20241228_214319](https://github.com/user-attachments/assets/79ec78f9-28a9-456a-bcf0-b63f6e12f609)

**Comment by victorchinyt** [comment]
Hi, I've also faced this issue as well and wanted to share my findings.

I believe the main cause of this issue (and also issue #12) was the combination of usernames used by the script with the current mail.tm domain `@freesourcecodes.com`. The usernames were running out, which made it appear as if the script wasn't working. To test this, I modified the script to accept manual username input, and when I tried an already-used username, the same issue happened (don't mind the extra logs I added). Also when running the unmodified script, the first account of the bulk would sometimes succeed, but all subsequent attempts would fail. This suggests that the first username might have been still available.

![Image](https://github.com/user-attachments/assets/3973e78d-2f6d-4a5b-aa26-b661df6068bc)

For context, the username combinations in this script are generated in the [pymailtm](https://github.com/CarloDePieri/pymailtm/blob/master/pymailtm/pymailtm.py) script using the `random_username` [package](https://pypi.org/project/random-username/), which creates combinations from a [list of predefined adjectives and nouns](https://github.com/williexu/random_username/tree/master/random_username/data), and a random number.

In my case, I edited this script to use a local edited `pymailtm.py`, which has said `random_username` and its list implemented. With this, I can edit the list for more different combinations, or have it use multiple numbers instead of just one. 

In short, this script is working properly, and should work now as well since mail.tm had rotated out the old domain. These are just my findings, but I hope they help!

**Comment by f-o** [comment]
> Hi, I've also faced this issue as well and wanted to share my findings.
> 
> I believe the main cause of this issue (and also issue [#12](https://github.com/f-o/MEGA-Account-Generator/issues/12)) was the combination of usernames used by the script with the current mail.tm domain `@freesourcecodes.com`. The usernames were running out, which made it appear as if the script wasn't working. To test this, I modified the script to accept manual username input, and when I tried an already-used username, the same issue happened (don't mind the extra logs I added). Also when running the unmodified script, the first account of the bulk would sometimes succeed, but all subsequent attempts would fail. This suggests that the first username might have been still available.


First of all, thank you so much for your investigation and detailed explanation @victorchinyt.
It seems you've identified a real bottleneck to this script, which I wasn't even aware of.

Some people seem to really be hammering out new accounts, which I never anticipated, and that's clearly affecting the overall stability of the script.
That being said, with the `random-username` package, it would seem we're only ever going to get 198.880 unique usernames regardless, which may have been enough 7 years ago (when the package was last updated), but isn't really gonna cut it now a days.

I've been neglecting this script for a while, since it already fulfilled its purpose for me, which was getting ~20 unique accounts, but given the fact that it has taken off, and is now being used by many other people, I might go back and fix some of the most blatant issues at hand. This is definitely one of the issues that needs to be addressed.

I'm thinking about taking the generation of usernames off from the package, and implement a new generator locally.
By bumping the lists from ~150 words each to something like 500, the script should be able to give a much more diverse range of unique usernames.
We can also add a few more digits, instead of just 1, which will expand the possibilities exponentially.

These things will only affect how long we're able to get unique addresses from Mail.tm, and not address the bigger issue at hand (in my opinion), which is Mega manually banning the current Mail.tm domain once in a blue moon.
My goal is still to implement multiple different temp mail backends, in a way so we can spread the load, and also prevent the script from just completely dying whenever the inevitable happens again.

---

All of this being said, I'm still cautious about making such a "fool-proof" script available to the public, and the implications this might bring to Mega.
Some people really like to abuse the service, and when Mega went out of their way to manually block the Mail.tm domain the first time, I realized this is having a real life impact on their service. I'm aware that there are other scripts doing this as well, but I'm still not sure how I feel about the ethical responsibilities of feeding into abusive behaviour. 

I can try to implement limits, throttles, and all those fancy things, but in the end if someone wants to create 100.000 accounts, and fill them with porn, they're gonna achieve this no matter what I try to do.

I really like Mega and their service, and I feel bad for enabling such blatant abuse, when they've indirectly said "no" (with their ban of current Mail.tm domain), but at the same time I'm just the author of a script, and maybe it isn't my job to police what people use it for, whether this be good or bad.
